apps/nextjs/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -32,8 +32,11 @@ export const GET = async (
     cookies().delete(EXPO_COOKIE_NAME);
 
     const authResponse = await DEFAULT_GET(req);
-    const setCookie = authResponse.headers.getSetCookie()[0];
+    const setCookie = authResponse.headers
+      .getSetCookie()
+      .find((cookie) => cookie.startsWith("authjs.session-token"));
```

Comment:
This doesn't work for me, as the cookie starts differently. Since the cookie pattern is defined above (and used just below that), could we just use it here to match?

```js
const setCookie = authResponse.headers
  .getSetCookie()
  .find((cookie) => AUTH_COOKIE_PATTERN.test(cookie));
```

Reply:
The pattern itself is hardcoded as well - if you change the authjs cookie name, this will have issues no matter what. Perhaps it might make more sense to customize the cookie to be `acme.session-token` instead, and that way users of the app can just find+replace `acme` and will cover the cookie as well?

An alternative would be to just ignore the prefix with something like this:
```ts
const AUTH_COOKIE_PATTERN = /(?:\w+)\.session-token=([^;]+)/;
```
And then apply your suggestion to test the cookie.

Reply:
@Wundero I just realised that I misscopied the contents of the cookie 🤦

The part of the `authjs.session-token` does not change (at least on my case), I just wanted to highlight that the  cookie contains another starting part (the  `authjs.pkce.code_verifier=...` which caused that the `.startsWith(...)` condition to not comply.

```
authjs.pkce.code_verifier=; Max-Age=0; Path=/; HttpOnly; SameSite=Lax, authjs.session-token=e9bf1a44-44f1-453f-b492-fc26adec22ef; Path=/; Expires=Sat, 22 Jun 2024 07:33:02 GMT; HttpOnly; SameSite=Lax
```

With that being clarified, your point is still valid as NextAuth allows to [configure cookies (including changing its names)](https://next-auth.js.org/configuration/options#cookies), so even the `.session-token` part could be changed. Is this something that needs to be supported? IMHO it's too of an edge case and not worth it, but still, adding a possible way below.

<details><summary>Details</summary>
<p>


I guess the regex could be created from exported `authConfig`'s cookie configuration (if any), defaulting to the current one:

```ts
const AUTH_COOKIE_NAME =
  authConfig.cookies?.sessionToken?.name || "authjs.session-token";
const AUTH_COOKIE_PATTERN = new RegExp(
  `${AUTH_COOKIE_NAME.replace(/\./g, "\\.")}=([^;]+)`,
);
```

Notes:
- because of the `satisfies NextAuthConfig`: the actual exported `authConfig` does not have the `cookies` and the nested properties, so TS coplains of the `authConfig.cookies?.sessionToken?.name` access, so this would need to be handled
- the `authConfig` object needs to be exported from `@acme/auth`
- there's some string manipulation things going on (replacing `.` with `\\.` so the regexp is properly created... there could be more edge cases so things could break

</p>
</details>

Reply:
The `startsWith` should still work, since the `getSetCookie` will return a list of all named cookies - the `authjs.pkce.code_verifier` (or any other authjs cookie) is a separate entry in the list, and should fail that check. In your example, it would split the cookies at the `,` and return these:
```ts
['authjs.pkce.code_verifier=; Max-Age=0; Path=/; HttpOnly; SameSite=Lax', 'authjs.session-token=e9bf1a44-44f1-453f-b492-fc26adec22ef; Path=/; Expires=Sat, 22 Jun 2024 07:33:02 GMT; HttpOnly; SameSite=Lax']
```

Reply:
I'm not why (maybe different system, code or package versions?), but In my case it does not return the same as you.

What you provided:
```js
// outer array
[
  // item 1
  'authjs.pkce.code_verifier=; Max-Age=0; Path=/; HttpOnly; SameSite=Lax',
  // item 2
  'authjs.session-token=e9bf1a44-44f1-453f-b492-fc26adec22ef; Path=/; Expires=Sat, 22 Jun 2024 07:33:02 GMT; HttpOnly; SameSite=Lax'
]
```

What I'm getting:
```js
// outer array
[
  // item 1 - the comma is inside the string -------------------------- v
  'authjs.pkce.code_verifier=; Max-Age=0; Path=/; HttpOnly; SameSite=Lax, authjs.session-token=e9bf1a44-44f1-453f-b492-fc26adec22ef; Path=/; Expires=Sat, 22 Jun 2024 07:33:02 GMT; HttpOnly; SameSite=Lax'
]
```

So, in my case there's a single item that does not start with `authjs.session-token`.

To add a bit more into the mix, I'm doing some tests with HTTPS and the name of the cookie changes, which would make the `startsWith` to not work even if your case (note the `__Secure-authjs.session-token` names). TBH I'm trying with `nextjs --experimental-https`, haven't tried this in a production site or anything, so not sure if that would affect:
```js
[
  '__Secure-authjs.pkce.code_verifier=; Max-Age=0; Path=/; HttpOnly; Secure; SameSite=Lax, __Secure-authjs.session-token=d99b92b9-acf2-4f01-b1bc-91cff0bff535; Path=/; Expires=Thu, 27 Jun 2024 07:33:38 GMT; HttpOnly; Secure; SameSite=Lax'
]
```

In any case, IMHO it feels more consistent and resilient doing the `AUTH_COOKIE_PATTERN.test(...)`, as just below it's doing a `setCookie?.match(AUTH_COOKIE_PATTERN)` which also allows the cookie to be in any place of the string.